devito/iteration.py

```diff
@@ -72,3 +111,18 @@ def signature(self):
         """
         signatures = [e.signature for e in self.expressions]
         return filter_ordered(chain(*signatures))
+
+    def indexify(self):
+        """Convert all enclosed stencil expressions to "indexed" format"""
+        for e in self.expressions:
+            e.indexify()
```

Comment:
Does this modify the expressions in place ? Maybe it's time to change this behaviour